rust/common/error/macros.rs

```diff
@@ -24,19 +24,19 @@
 macro_rules! error_messages {
     {
         $name:ident code: $code_pfx:literal, type: $message_pfx:literal,
-        $($error_name:ident( $($inner:ty),* $(,)? ) = $code:literal: $body:literal),+ $(,)?
+        $($error_name:ident $({ $($field:ident : $inner:ty),+ $(,)? })? = $code:literal: $body:literal),+ $(,)?
     } => {
```

Comment:
Error messages are now struct variants rather than tuple variants. This change allows us to name the payload, refer to it by name in the formatting string, and greatly simplifies this macro.